pkg/ansible/proxy/proxy_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	kcorev1 "k8s.io/api/core/v1"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func TestHandler(t *testing.T) {
```

Comment:
I think that instead of running the proxy, I think using this package would be the correct approach: https://golang.org/pkg/net/http/httptest/ to test the server and request

I would also like to see some tests using a fake cache and testing the correct logic.

https://godoc.org/k8s.io/client-go/tools/cache/testing#FakeControllerSource
and 
https://godoc.org/sigs.k8s.io/controller-runtime/pkg/cache/informertest#FakeInformers


Lastly, if the test cases could look like the [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests)

Reply:
I'm having a lot of trouble integrating this. Will speak with you tomorrow about it.